<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModelWindow</class>
 <widget class="QMainWindow" name="ModelWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>873</width>
    <height>447</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TSim</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="BlockGraphicsView" name="block_graphics"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>873</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionFileNew"/>
    <addaction name="separator"/>
    <addaction name="actionFileOpen"/>
    <addaction name="actionFileClose"/>
    <addaction name="separator"/>
    <addaction name="actionFileSave"/>
    <addaction name="actionFileSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuBlocks">
    <property name="title">
     <string>Blocks</string>
    </property>
    <addaction name="actionLibrary"/>
    <addaction name="actionBlockRemove"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuModel">
    <property name="title">
     <string>Model</string>
    </property>
    <addaction name="actionModelUpdate"/>
    <addaction name="actionModelBuildExecutor"/>
    <addaction name="actionModelShowDiagnostics"/>
    <addaction name="actionModelParameters"/>
    <addaction name="separator"/>
    <addaction name="actionModelGenerateCode"/>
   </widget>
   <widget class="QMenu" name="menuSim">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="title">
     <string>Sim</string>
    </property>
    <addaction name="actionSimStep"/>
    <addaction name="actionSimReset"/>
    <addaction name="actionSimShowPlot"/>
    <addaction name="actionSimClear"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuBlocks"/>
   <addaction name="menuModel"/>
   <addaction name="menuSim"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QToolBar" name="toolBar_2">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionBlockAddLimiter">
   <property name="text">
    <string>Add Limiter</string>
   </property>
  </action>
  <action name="actionFileOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionFileSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionFileSaveAs">
   <property name="text">
    <string>Save As</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionBlockRemove">
   <property name="text">
    <string>Remove Selected Block</string>
   </property>
  </action>
  <action name="actionBlockAddSin">
   <property name="text">
    <string>Add SIN</string>
   </property>
  </action>
  <action name="actionBlockAddClock">
   <property name="text">
    <string>Add Clock</string>
   </property>
  </action>
  <action name="actionModelUpdate">
   <property name="text">
    <string>Update</string>
   </property>
  </action>
  <action name="actionModelBuildExecutor">
   <property name="text">
    <string>Build Executor</string>
   </property>
  </action>
  <action name="actionSimStep">
   <property name="text">
    <string>Step</string>
   </property>
  </action>
  <action name="actionSimReset">
   <property name="text">
    <string>Reset</string>
   </property>
  </action>
  <action name="actionSimShowPlot">
   <property name="text">
    <string>Show Plot</string>
   </property>
  </action>
  <action name="actionLibrary">
   <property name="text">
    <string>Block Library</string>
   </property>
  </action>
  <action name="actionSimClear">
   <property name="text">
    <string>Clear</string>
   </property>
  </action>
  <action name="actionModelShowDiagnostics">
   <property name="text">
    <string>Show Diagnostics</string>
   </property>
  </action>
  <action name="actionModelParameters">
   <property name="text">
    <string>Parameters</string>
   </property>
  </action>
  <action name="actionFileClose">
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actionFileNew">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionModelGenerateCode">
   <property name="text">
    <string>Generate Code</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>BlockGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>widgets/block_graphics_view.h</header>
   <slots>
    <slot>removeSelectedBlock()</slot>
    <slot>updateModel()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>315</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLibrary</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>showLibrary()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionModelUpdate</sender>
   <signal>triggered()</signal>
   <receiver>block_graphics</receiver>
   <slot>updateModel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>315</x>
     <y>162</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionModelBuildExecutor</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>generateExecutor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSimClear</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>clearExecutor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSimStep</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>stepExecutor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSimShowPlot</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>showPlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionBlockRemove</sender>
   <signal>triggered()</signal>
   <receiver>block_graphics</receiver>
   <slot>removeSelectedBlock()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>315</x>
     <y>162</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionModelShowDiagnostics</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>showDiagnostics()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFileSave</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>saveModel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>315</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFileOpen</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>openModel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>315</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionModelParameters</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>showModelParameters()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFileSaveAs</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>saveModelAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSimReset</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>resetExecutor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFileClose</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>closeModel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFileNew</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>newModel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionModelGenerateCode</sender>
   <signal>triggered()</signal>
   <receiver>ModelWindow</receiver>
   <slot>saveCode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>saveModel()</slot>
  <slot>openModel()</slot>
  <slot>saveModelAs()</slot>
  <slot>showDiagnostics()</slot>
  <slot>generateExecutor()</slot>
  <slot>stepExecutor()</slot>
  <slot>clearExecutor()</slot>
  <slot>showLibrary()</slot>
  <slot>showModelParameters()</slot>
  <slot>showPlot()</slot>
  <slot>addBlock(QString,QString)</slot>
  <slot>resetExecutor()</slot>
  <slot>closeModel()</slot>
  <slot>newModel()</slot>
  <slot>saveCode()</slot>
 </slots>
</ui>
